osg/short-term.md

```diff
@@ -0,0 +1,292 @@
+file. Once you have the ticket and path, `iget` can be executed as follows:
+
+```
+iget -rt <ticket> "<path>"
+```
+
+Note: if you're using a command shell to execute `iget` then the quotes around the path are important if the path
+contains whitespace. In cases where the wrapper script is not a UNIX shell, the arguments can usually explicitly be
+passed in as a list. For example, this function can be used in Python:
+
+``` python
+# Download a file or directory from iRODS.
+def download_file(ticket, src):
+    rc = subprocess.call(["iget", "-rt", ticket, src])
+    if rc != 0:
+        raise Exception("could not download {0}".format(src))
+```
+
+## Running the Job
+
+Once the input files have been downloaded, the job itself can be executed. In many cases, the job is executed as a
+subprocess of the wrapper script. This is a useful practice because it provides a clear boundary between the code that
+interacts with the DE and the code that performs the analysis itself. This Python example simply calles `wc` passing in
+the arguments retrieved from the configuration file:
+
+``` python
+# Run the word count job.
+def run_job(arguments, output_filename, error_filename):
+    with open(output_filename, "w") as out, open(error_filename, "w") as err:
+        rc = subprocess.call(["wc"] + arguments, stdout=out, stderr=err)
+        if rc != 0:
+            raise Exception("wc returned exit code {0}".format(rc))
+```
+
+## Uploading Output Files
+
+Uploading output files is a little bit more complicated than retrieving input files because additional steps have to be
+taken to ensure that the user who ran the job can view the output files. The steps are:
+
+- upload the files
+- give the user ownership of the file if the iRODS user is different from the user who submitted the job
+- remove the iRODS user's privileges on the file if the iRODS user is different from the user who submitted the job
+
+The following example is from the [wrapper script reference implementation](2):
+
+``` python
+# Upload a file or directory to iRODS.
+def upload_file(ticket, irods_user, owner, src, dest):
+    rc = subprocess.call(["iput", "-rt", ticket, src, dest])
+    if rc != 0:
+        raise Exception("could not upload {0} to {1}".format(src, dest))
+
+    # Don't bother modifying the file permissions if the iRODS user and file owner are the same.
+    if irods_user == owner:
+        return
+
+    # Update the file permissions so that the person who submitted the job can see the output files.
+    fullpath = os.path.join(dest, src)
+    rc = subprocess.call(["ichmod", "own", owner, fullpath])
+    if rc != 0:
+        raise Exception("could not change the owner of {0} to {1}".format(fullpath, owner))
+    rc = subprocess.call(["ichmod", "null", irods_user, fullpath])
+    if rc != 0:
+        raise Exception("could not remove {0} permissions on {1}".format(irods_user, fullpath))
```

Comment:
Nitpicky, but we might want to ensure that everything at least makes an attempt to upload and to change ownership/drop permissions, so I guess the `upload_files` example below should maybe specifically catch errors? Again, just being a bit paranoid that our examples/documentation are as close a reflection of best practices as we can muster, heh